src/debugger/adapter/DebugAdapter.js

```diff
@@ -126,9 +128,17 @@ class PrepackDebugSession extends DebugSession {
         process.exit();
       },
     };
+
+    this._ensureAdapterChannelCreated("launchRequest");
+    invariant(this._adapterChannel !== undefined, "Adapter Channel used before it was created, in debugger.");
     this._adapterChannel.launch(response.request_seq, launchArgs, (dbgResponse: DebuggerResponse) => {
       this.sendResponse(response);
     });
+
+    // The UI will end the configuration sequence by calling 'configurationDone' request.
+    // Sending this during the initializeResponse caused non-determinism in request ordering
```

Comment:
This comment states the negative side, and assumes some history context which reader may be confused. 
Instead, I would change to:
```
Important: Responding InitializedEvent is the master switch that caused UI to start sending further requests like breakpoints, execution control etc... So InitializedEvent should be sent after  _adapterChannel is ready and prepack is launched so that we are ready to handle these requests.
```